### ISSUE TITLE
incompatible_msrv fails to detect std::iter::repeat_n

### ISSUE THREAD

**Issue by akonradi-signal** [rust-lang/rust-clippy#14212]
### Summary

[`std::iter::repeat_n`](https://doc.rust-lang.org/stable/std/iter/fn.repeat_n.html) was stabilized in Rust 1.82.0. Running clippy with an earlier MSRV does not detect the incompatible call.

### Lint Name

incompatible_msrv

### Reproducer

I tried this code:

```rust
#[clippy::msrv = "1.80.0"]
#[warn(clippy::incompatible_msrv)]
fn repeat_n() {
    let _ = std::iter::repeat_n((), 5);
}
```

I expected to see this happen: a warning about `repeat_n` not being stable until 1.82.0

Instead, this happened: nothing

### Version

```text
rustc 1.86.0-nightly (a9730c3b5 2025-02-05)
binary: rustc
commit-hash: a9730c3b5f84a001c052c60c97ed0765e9ceac04
commit-date: 2025-02-05
host: x86_64-unknown-linux-gnu
release: 1.86.0-nightly
LLVM version: 19.1.7
```

**Comment by samueltardieu** [comment]
I may be misunderstanding what you're saying, but using `#[clippy::msrv = "1.80.0"]` does mean "Do not suggest things that will not be compatible with Rust 1.80.0". It does mean "Signal code invalid in Rust 1.80.0", as signaling invalid code is the job of the compiler, not the job of Clippy.

**Comment by akonradi-signal** [comment]
I'd expect a compiler version earlier than 1.82 to fail to compile the above code. I'd expect one later than 1.82.0 to compile it without any warnings since the function is defined. I would still expect clippy to warn about the use of `repeat_n`.

I filed this bug because when I tried the same formula with other recently-defined standard library functions, clippy complained as expected. I'd expect this code to produce 2 warnings:
```rust
#![warn(clippy::incompatible_msrv)]

#[clippy::msrv = "1.80"]
pub fn requires_rust_1_82() {
    let _ = std::iter::repeat_n((), 5);
}

#[clippy::msrv = "1.75"]
pub fn requires_rust_1_79() {
    let _ = std::path::absolute("asdf");
}
```
but running clippy on the [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=36491b6ce5ac9cfe23671ef8e2c5dd36) only produces 1:
```
warning: current MSRV (Minimum Supported Rust Version) is `1.75.0` but this item is stable since `1.79.0`
  --> src/lib.rs:10:13
   |
10 |     let _ = std::path::absolute("asdf");
   |             ^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#incompatible_msrv
note: the lint level is defined here
  --> src/lib.rs:1:9
   |
1  | #![warn(clippy::incompatible_msrv)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `playground` (lib) generated 1 warning
```

**Comment by samueltardieu** [comment]
My bad, you are right. I overlooked the `incompatible_msrv` lint there.

@rustbot claim
